internal/events/submanager.go

```diff
@@ -108,6 +113,12 @@ func NewSubscriptionManager(conf *SubscriptionManagerConf, rpc eth.RPCClient, ws
 	if conf.EventPollingIntervalSec <= 0 {
 		conf.EventPollingIntervalSec = 1
 	}
+	if conf.CatchupModeBlockGap <= 0 {
```

Comment:
Nit.. Could we change the type of `CatchupModeBlockGap` to `uint` so there's no need to check for negative values?

Reply:
Thanks @vdamle - when I went through and tried making all the changes for this, I found the downstream complexity of casting and dealing with mapping to `big.Int` with comparison, became much more complicated than leaving it as an `int64`. So if it's ok, I'm going to leave this one as is.